lib/utils/FileCache.mjs

```diff
@@ -59,24 +59,59 @@ export default class FileCache {
 	hasFileChanged(absoluteFilepath) {
 		// Get file descriptor compares current metadata against cached
 		// one and stores the result to "changed" prop.w
+
+		const fileCheckSumChanged = this._useCheckSum
+			? this.hasFileCheckSumChanged(absoluteFilepath)
+			: false;
 		const descriptor = this._fileCache.getFileDescriptor(absoluteFilepath);
-		/** @type {CacheMetadata} */
-		const meta = descriptor.meta || {};
-		const changed = descriptor.changed || meta.hashOfConfig !== this._hashOfConfig;
+
+		descriptor.meta.data ??= {};
+
+		assert(isPlainObject(descriptor.meta.data));
+
+		const configChanged = descriptor.meta.data.hashOfConfig !== this._hashOfConfig;
+
+		let changed = false;
+
+		if (this._useCheckSum) {
+			changed = configChanged || fileCheckSumChanged;
```

Comment:
[suggestion] Since `getFileDescriptor()` already calculates a file hash if `useCheckSum` is true, `#hasFileCheckSumChanged()` seems to be simplified like this:

https://github.com/jaredwray/cacheable/blob/5ec1858bdd07aa5efc30e87f14253a951fbc83b5/packages/file-entry-cache/src/index.ts#L267-L271

```suggestion
			changed = configChanged || this.#hasFileCheckSumChanged(descriptor.key, descriptor.meta.hash);
```

```js
	/**
	 * @param {string} key
	 * @param {string | undefined} fileHash
	 * @returns {boolean}
	 */
	#hasFileCheckSumChanged(key, fileHash) {
		if (!fileHash) {
			return true;
		}

		const metaCache = this._fileCache.cache.getKey(key);

		if (fileHash !== metaCache?.hash) {
			return true;
		}

		return false;
	}
```

Reply:
Thank you for this!

I suspected there might be a way to simplify by reusing the data already in `descriptor` but didn't quite see it yesterday :)

Much better like this 🙇